src/services/doppler-api-client.ts

```diff
@@ -143,6 +143,57 @@ export class HttpDopplerApiClient implements DopplerApiClient {
     }));
   }
 
+  private getunsubscribedStatus(subscriber: any): string {
+    let unsubscribedStatus: string = '';
+    switch (subscriber.unsubscriptionType) {
+      case 'hardBounce':
+        unsubscribedStatus = 'unsubscribed_by_hard';
+        break;
+      case 'softBounce':
+        unsubscribedStatus = 'unsubscribed_by_soft';
+        break;
+      case 'neverOpen':
+        unsubscribedStatus = 'unsubscribed_by_never_open';
+        break;
+      case 'manual':
+        unsubscribedStatus =
+          subscriber.manualUnsubscriptionReason === 'administrative'
+            ? 'unsubscribed_by_client'
+            : 'unsubscribed_by_subscriber';
+        break;
+      case 'abuseLink':
+      case 'feedbackLoop':
+      case 'internalPolicies':
+        unsubscribedStatus = 'unsubscribed_by_subscriber';
+        break;
+      default:
+        unsubscribedStatus = '';
+        break;
+    }
+    return unsubscribedStatus;
+  }
+
+  private getSubscriberStatus(subscriber: any): string {
```

Comment:
Here just 3 params would be nicer I think
```suggestion
  private getSubscriberStatus(status: string, unsubscriptionType: string,manualUnsubscriptionReason: string): string {
```